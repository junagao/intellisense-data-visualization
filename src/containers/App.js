import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader/root';
import { connect } from 'react-redux';

import { getData, setHover, setLegendPosition } from 'actions/data';
import { Table, LineGraph } from 'components';
import AppContainer from 'containers/App.styles';

export class App extends React.Component {
  componentDidMount() {
    const { getData } = this.props;
    getData();
  }

  render() {
    const {
      data,
      setHover,
      hoverLineGraph,
      hoveredMetric,
      hoveredMetricColor,
      setLegendPosition,
      legendXPosition,
      legendYPosition,
      selectedMetrics,
    } = this.props;

    return (
      <AppContainer>
        <Table data={data} />
        <LineGraph
          data={data}
          onLineHover={setHover}
          hoverLineGraph={hoverLineGraph}
          hoveredMetric={hoveredMetric}
          hoveredMetricColor={hoveredMetricColor}
          onLegendPosition={setLegendPosition}
          legendXPosition={legendXPosition}
          legendYPosition={legendYPosition}
          selectedMetrics={selectedMetrics}
        />
      </AppContainer>
    );
  }
}

const mapStateToProps = (state) => ({
  data: state.data,
  hoverLineGraph: state.hoverLineGraph,
  hoveredMetric: state.hoveredMetric,
  hoveredMetricColor: state.hoveredMetricColor,
  legendXPosition: state.legendXPosition,
  legendYPosition: state.legendYPosition,
  selectedMetrics: state.selectedMetrics,
});

const mapDispatchToProps = {
  getData,
  setHover,
  setLegendPosition,
};

App.propTypes = {
  getData: PropTypes.func.isRequired,
  data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  setHover: PropTypes.func.isRequired,
  hoverLineGraph: PropTypes.bool.isRequired,
  hoveredMetric: PropTypes.string.isRequired,
  hoveredMetricColor: PropTypes.string.isRequired,
  setLegendPosition: PropTypes.func.isRequired,
  legendXPosition: PropTypes.number.isRequired,
  legendYPosition: PropTypes.number.isRequired,
  selectedMetrics: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export const connectedApp = hot(
  connect(mapStateToProps, mapDispatchToProps)(App),
);
