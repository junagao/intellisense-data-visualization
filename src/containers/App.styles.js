import styled from 'styled-components';

const AppContainer = styled.div`
  padding: 1rem;
  margin: 0 auto;
  max-width: 50rem;
`;

export default AppContainer;
