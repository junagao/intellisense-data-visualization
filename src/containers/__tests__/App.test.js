import React from 'react';
import { mount } from 'enzyme';

import { App } from 'containers/App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<App getData={jest.fn()} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
