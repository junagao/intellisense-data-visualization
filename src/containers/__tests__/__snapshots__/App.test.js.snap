// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<App
  getData={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div>
    Intellisense
  </div>
</App>
`;
